packages/ui/src/composables/useClearable.ts

```diff
@@ -0,0 +1,47 @@
+import { computed, Ref, PropType } from 'vue'
+
+interface ClearableProps {
+  clearable: boolean
+  clearValue: any
+  disabled?: boolean
+  readonly?: boolean
+  multiple?: boolean
+  success?: boolean
+  color?: string
+}
+
+export const useClearableProps = {
+  clearable: { type: Boolean, default: false },
+  clearableIcon: { type: String, default: 'highlight_off' },
+  clearValue: { type: String as PropType<any>, default: '' },
+}
+
+export const useClearableEmits = ['clear']
+
+export const useClearable = (
+  props: ClearableProps,
+  inputValue: Ref<any>,
+  isFocused: Ref<boolean>,
+  hasError: Ref<boolean>,
+) => {
+  const clearedValues = [null, undefined, props.clearValue]
+
+  const canBeCleared = computed(() => (
+    props.clearable &&
+    !props.disabled &&
+    !clearedValues.includes(inputValue.value)
```

Comment:
I fixed it for `va-time-input` in the component.
I didn't change the `useClearable` hook, because the `va-input` component (that also uses this hook) has to display the `clear` icon when it is used in other components.

Reply:
I believe VaInput component should work as other components. When VaInput used in other component, we need to use VaInputWrapper with a prop doShowClearIcon. VaInput's logic mustn't change for other components. It must be independent,

Otherwise, we will have different logic for same feature in different components.

Reply:
I corrected `useClearable` hook to handle `readonly` props as well as `disabled`.